source/acquaman/AMGenericContinuousScanConfiguration.cpp

```diff
@@ -0,0 +1,328 @@
+		connect(scanAxisAt(0)->regionAt(0), SIGNAL(regionEndChanged(AMNumber)), this, SLOT(computeTotalTime()));
+		connect(scanAxisAt(0)->regionAt(0), SIGNAL(regionTimeChanged(AMNumber)), this, SLOT(computeTotalTime()));
+	}
+
+	else if (axisId == 0){
+
+		axisControlInfos_.replace(0, newInfo);
+		setModified(true);
+	}
+
+	else if (axisId == 1 && axisControlInfos_.count() == 1){
+
+		axisControlInfos_.append(newInfo);
+		setModified(true);
+
+		AMScanAxisRegion *region = new AMScanAxisRegion;
+		AMScanAxis *axis = new AMScanAxis(AMScanAxis::StepAxis, region);
+		appendScanAxis(axis);
+
+		connect(scanAxisAt(1)->regionAt(0), SIGNAL(regionStartChanged(AMNumber)), this, SLOT(computeTotalTime()));
+		connect(scanAxisAt(1)->regionAt(0), SIGNAL(regionStepChanged(AMNumber)), this, SLOT(computeTotalTime()));
+		connect(scanAxisAt(1)->regionAt(0), SIGNAL(regionEndChanged(AMNumber)), this, SLOT(computeTotalTime()));
+		connect(scanAxisAt(1)->regionAt(0), SIGNAL(regionTimeChanged(AMNumber)), this, SLOT(computeTotalTime()));
+	}
+
+	else if (axisId == 1){
+
+		axisControlInfos_.replace(1, newInfo);
+		setModified(true);
+	}
+
+	computeTotalTime();
+}
+
+void AMGenericContinuousScanConfiguration::removeControl(int axisId)
+{
+	if (axisId < axisControlInfos_.count() && axisId >= 0){
+
+		AMScanAxis *axis = removeScanAxis(axisId);
+		axis->regionAt(0)->disconnect();
+		axis->regionAt(0)->deleteLater();
+		axis->deleteLater();
+		axisControlInfos_.remove(axisId);
+		setModified(true);
+	}
+}
+
+void AMGenericContinuousScanConfiguration::addDetector(AMDetectorInfo newInfo)
+{
+	bool containsDetector = false;
+
+	for (int i = 0, size = detectorConfigurations_.count(); i < size; i++){
+
+		if (newInfo.name() == detectorConfigurations_.at(i).name())
+			containsDetector = true;
+	}
+
+	if (!containsDetector){
+
+		detectorConfigurations_.append(newInfo, newInfo.name());
+		setModified(true);
+	}
+}
+
+void AMGenericContinuousScanConfiguration::removeDetector(AMDetectorInfo info)
+{
+	bool detectorRemoved = false;
+
+	for (int i = 0, size = detectorConfigurations_.count(); i < size && !detectorRemoved; i++){
+
+		if (info.name() == detectorConfigurations_.at(i).name()){
+
+			detectorRemoved = true;
+			detectorConfigurations_.remove(i);
+			setModified(true);
+		}
+	}
+}
+
+void AMGenericContinuousScanConfiguration::addRegionOfInterest(AMRegionOfInterest *region)
+{
+	regionsOfInterest_.append(region);
+	setModified(true);
+}
+
+void AMGenericContinuousScanConfiguration::removeRegionOfInterest(AMRegionOfInterest *region)
+{
+	foreach (AMRegionOfInterest *regionToBeRemoved, regionsOfInterest_)
+		if (regionToBeRemoved->name() == region->name()){
+
+			regionsOfInterest_.removeOne(regionToBeRemoved);
+			setModified(true);
+		}
+}
+
+void AMGenericContinuousScanConfiguration::setI0(const AMDetectorInfo &info)
+{
+	bool nameInDetectorList = false;
+
+	for (int i = 0, size = detectorConfigurations_.count(); i < size && !nameInDetectorList; i++)
```

Comment:
This might be the body of a cool method to provide all scan configurations, a hasDetector(const QString&) type method.